### ISSUE TITLE
Does pip install ultralytics supports jetpack 6.2?

### ISSUE THREAD

**Issue by lintong-zhang** [ultralytics/ultralytics#21487]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi

I am running jetpack 6.2 on nvidia jetson orin NX. 

I tried to follow instructions with 6.1 in hoping it would work for 6.2. I am able to install torch 2.5.0 wheel and torch vision 0.20.0. It works when I test `import torch` and `is_cuda_avalible` in python. Please see the following `jetson_release` results. 

The problem is when I do `pip installl ultralytics`, it never finishes and constantly goes into `multiple version of package X found, trying to resolve conflict ... it takes longer than expected...`

My question is if anyone has working version of ultralytics with jetson 6.2. Or does the instruction indeed work for 6.2? If not, what should i do? 

@glenn-jocher You seem to know and handle this part.

### Additional

jetson_release 
Software part of jetson-stats 4.3.2 - (c) 2024, Raffaello Bonghi
Model: CTI Boson for Orin  + Orin NX - Jetpack 6.2 [L4T 36.4.3]
NV Power Mode[0]: MAXN
Serial Number: [XXX Show with: jetson_release -s XXX]
Hardware:
 - P-Number: p3767-0000
 - Module: NVIDIA Jetson Orin NX (16GB ram)
Platform:
 - Distribution: Ubuntu 22.04 Jammy Jellyfish
 - Release: 5.15.148-tegra
jtop:
 - Version: 4.3.2
 - Service: Active
Libraries:
 - **CUDA: 12.6.68**
 - **cuDNN: 9.3.0.75**
 - TensorRT: 10.3.0.30
 - VPI: 3.2.4
 - OpenCV: 4.5.4 - with CUDA: NO

**Comment by UltralyticsAssistant** [comment]
👋 Hello @lintong-zhang, thank you for reaching out and providing detailed information about your setup! 🚀 This is an automated response to help you get started, and an Ultralytics engineer will assist you soon.

We recommend reviewing the [Docs](https://docs.ultralytics.com/) for best practices, especially regarding [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage. For Jetson-specific questions, you may find related discussions and solutions in the [issues](https://github.com/ultralytics/ultralytics/issues) or [discussions](https://github.com/orgs/ultralytics/discussions) sections.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug further.

If you are seeking advice for custom training or deployment, please include as much information as possible about your environment, error logs, and steps you've followed.

Join the Ultralytics community for real-time support and peer help:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for chat
- [Discourse](https://community.ultralytics.com/) for in-depth topics
- [Subreddit](https://www.reddit.com/r/Ultralytics/) for community discussions

## Upgrade

To ensure compatibility and access to the latest features, upgrade to the newest `ultralytics` package and verify all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) are installed in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these verified environments (all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) are preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your patience while we look into your question! If you have additional logs or details (such as the full pip output or specific errors encountered), please share them to help us assist you more effectively.

**Comment by Y-T-G** [comment]
You can try using

`pip install ultralytics --no-deps`

**Comment by nikunjlad** [comment]
@lintong-zhang `pip install ultralytics` works well for me, although I tested in on Jetpack 6.0 with Orin NX 16GB board. Which version of Python are you using? I am using Python3.12. Below are my steps:

1. First install python3.12 virtualenv: `python3.12 -m venv test`. This creates a test environment.
2. Activate the environment: `source test/bin/activate`
3. Just run -> `pip install ultralytics`. This should install all dependencies along with it. 

If you run using `--no-deps` flag in pip install as @Y-T-G mentioned above it still works (I verified it), but then you need to figure out the dependencies manually since ultralytics won't work without its needed dependencies. 

Things I would suggest you to do:
1. Install Python3.12 on Jetpack 6.2 (its been a while I worked with JP6.2 so not sure if this is shipped by default. I guess the default is Python3.10)
2. Install and setup a virtualenv: This isolates and prevents installation in system location and directories avoiding any permission issues.
3. Install ultralytics without any flags and let it figure out the dependencies. It should work well.

**Comment by lintong-zhang** [comment]
Thank you both and sorry for the late confirmation.
Yes, I can confirm that `pip install ultralytics --no-deps` works for jetpack 6.2. With out `--no-deps`, pip goes into a dependency hell. 
For anyone else want to use yolo in jetpack 6.2, the combination in the original post works.

**Comment by glenn-jocher** [comment]
Thanks for the update—JP6.2 is supported, and the Jetson guide has been tested on JP6.2; it also covers the cuSPARSELt fix for `torch==2.5.0` and installing the aarch64 `onnxruntime-gpu` wheel (plus the numpy pin). See the Quick Start Guide: NVIDIA Jetson with Ultralytics YOLO11 for details.

If you’d like to avoid pip resolution issues entirely, you can use the JetPack 6 Docker image:

```bash
t=ultralytics/ultralytics:latest-jetson-jetpack6
sudo docker pull $t && sudo docker run -it --ipc=host --runtime=nvidia $t
```

If everything’s working now, please close the issue; otherwise share `yolo checks` output and the full pip resolver logs as text for us to review.
